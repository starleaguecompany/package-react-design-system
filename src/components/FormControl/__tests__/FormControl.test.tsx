import * as React from 'react'
import { render, screen } from '@testing-library/react'

import { FormControl, TextInput } from '../../../index'

describe('FormControl', () => {
  test('should render correctly with no props', () => {
    render(<FormControl />)

    expect(screen.getByTestId('FormControl')).toMatchSnapshot()
    expect(screen.getByTestId('FormControl')).toHaveClass('container', { exact: true })
  })

  test('should render correctly with attributes', () => {
    render(<FormControl id="test-id" className="test-class" />)

    expect(screen.getByTestId('FormControl')).toMatchSnapshot()
    expect(screen.getByTestId('FormControl')).toHaveClass('test-class')
    expect(screen.getByTestId('FormControl')).toHaveAttribute('id', 'test-id')
  })

  test('should render correctly with children', () => {
    render(
      <FormControl>
        <TextInput placeholder="Test placeholder" />
        <FormControl.HelperText>Test helper text</FormControl.HelperText>
      </FormControl>
    )

    expect(screen.getByTestId('FormControl')).toMatchSnapshot()
    expect(screen.getByRole('textbox')).toBeVisible()
    expect(screen.getByText('Test helper text')).toBeVisible()
    expect(screen.getByText('Test helper text')).toHaveClass('helperText')
  })

  test('should render correctly with invalid prop', () => {
    render(
      <FormControl invalid>
        <TextInput placeholder="Test placeholder" />
        <FormControl.HelperText>Test helper text</FormControl.HelperText>
      </FormControl>
    )

    expect(screen.getByTestId('FormControl')).toMatchSnapshot()
    expect(screen.getByRole('textbox')).toHaveClass('invalid')
    expect(screen.getByTestId('TextInput')).toHaveClass('invalid')
    expect(screen.getByText('Test placeholder')).toHaveClass('invalid')
    expect(screen.getByText('Test helper text')).toHaveClass('invalid')
  })

  test('should render correctly with disabled prop', () => {
    render(
      <FormControl disabled>
        <TextInput placeholder="Test placeholder" />
        <FormControl.HelperText>Test helper text</FormControl.HelperText>
      </FormControl>
    )

    expect(screen.getByTestId('FormControl')).toMatchSnapshot()
    expect(screen.getByRole('textbox')).toHaveClass('disabled')
    expect(screen.getByTestId('TextInput')).toHaveClass('disabled')
    expect(screen.getByText('Test placeholder')).toHaveClass('disabled')
    expect(screen.getByText('Test helper text')).toHaveClass('disabled')
  })

  test('should change disabled attribute', () => {
    const { rerender } = render(
      <FormControl>
        <TextInput placeholder="Test placeholder" />
        <FormControl.HelperText>Test helper text</FormControl.HelperText>
      </FormControl>
    )

    expect(screen.getByTestId('FormControl')).toMatchSnapshot()
    expect(screen.getByRole('textbox')).not.toHaveClass('disabled')
    expect(screen.getByTestId('TextInput')).not.toHaveClass('disabled')
    expect(screen.getByText('Test placeholder')).not.toHaveClass('disabled')
    expect(screen.getByText('Test helper text')).not.toHaveClass('disabled')

    rerender(
      <FormControl disabled>
        <TextInput placeholder="Test placeholder" />
        <FormControl.HelperText>Test helper text</FormControl.HelperText>
      </FormControl>
    )

    expect(screen.getByRole('textbox')).toHaveClass('disabled')
    expect(screen.getByTestId('TextInput')).toHaveClass('disabled')
    expect(screen.getByText('Test placeholder')).toHaveClass('disabled')
    expect(screen.getByText('Test helper text')).toHaveClass('disabled')
  })
})
