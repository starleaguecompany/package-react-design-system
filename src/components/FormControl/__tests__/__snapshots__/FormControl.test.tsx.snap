// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormControl should change disabled attribute 1`] = `
<div
  class="container"
  data-qa="FormControl"
>
  <div
    class="container"
    data-qa="TextInput"
  >
    <div
      class="inputSegment"
    >
      <div
        class="inputWrapper"
      >
        <div
          class="inputValue"
        />
        <input
          autocomplete="off"
          class="input"
          spellcheck="false"
          type="text"
          value=""
        />
      </div>
    </div>
    <label
      class="placeholder"
    >
      Test placeholder
    </label>
  </div>
  <div
    class="helperText"
  >
    Test helper text
  </div>
</div>
`;

exports[`FormControl should render correctly with attributes 1`] = `
<div
  class="test-class container"
  data-qa="FormControl"
  id="test-id"
/>
`;

exports[`FormControl should render correctly with children 1`] = `
<div
  class="container"
  data-qa="FormControl"
>
  <div
    class="container"
    data-qa="TextInput"
  >
    <div
      class="inputSegment"
    >
      <div
        class="inputWrapper"
      >
        <div
          class="inputValue"
        />
        <input
          autocomplete="off"
          class="input"
          spellcheck="false"
          type="text"
          value=""
        />
      </div>
    </div>
    <label
      class="placeholder"
    >
      Test placeholder
    </label>
  </div>
  <div
    class="helperText"
  >
    Test helper text
  </div>
</div>
`;

exports[`FormControl should render correctly with disabled prop 1`] = `
<div
  class="container"
  data-qa="FormControl"
>
  <div
    class="container disabled"
    data-qa="TextInput"
  >
    <div
      class="inputSegment"
    >
      <div
        class="inputWrapper"
      >
        <div
          class="inputValue"
        />
        <input
          autocomplete="off"
          class="input disabled"
          disabled=""
          spellcheck="false"
          type="text"
          value=""
        />
      </div>
    </div>
    <label
      class="placeholder disabled"
    >
      Test placeholder
    </label>
  </div>
  <div
    class="helperText disabled"
  >
    Test helper text
  </div>
</div>
`;

exports[`FormControl should render correctly with invalid prop 1`] = `
<div
  class="container"
  data-qa="FormControl"
>
  <div
    class="container invalid"
    data-qa="TextInput"
  >
    <div
      class="inputSegment"
    >
      <div
        class="inputWrapper"
      >
        <div
          class="inputValue"
        />
        <input
          autocomplete="off"
          class="input invalid"
          spellcheck="false"
          type="text"
          value=""
        />
      </div>
    </div>
    <label
      class="placeholder invalid"
    >
      Test placeholder
    </label>
  </div>
  <div
    class="helperText invalid"
  >
    Test helper text
  </div>
</div>
`;

exports[`FormControl should render correctly with no props 1`] = `
<div
  class="container"
  data-qa="FormControl"
/>
`;
